File: MediCareWebSpa/src/app/DTOs/models/rolePermission.ts
import { Permission } from "./permission";
import { Role } from "./role";

export class RolePermission {
  id: number;
  roleId: number;
  role: Role;
  permissionId: number;
  permission: Permission;

  constructor(
    id: number,
    roleId: number,
    role: Role,
    permissionId: number,
    permission: Permission
  ) {
    this.id = id;
    this.roleId = roleId;
    this.role = role;
    this.permissionId = permissionId;
    this.permission = permission;
  }
}

File: MediCareWebApi/Models/RolePermission.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace MediCare.Models
{
	public class RolePermission
	{
		[Key]
		[Column(Order = 1)]
		public int RoleId { get; set; }

		[ForeignKey("RoleId")]
		public Role Role { get; set; }

		[Key]
		[Column(Order = 2)]
		public int PermissionId { get; set; }

		[ForeignKey("PermissionId")]
		public Permission Permission { get; set; }
	}
}
